x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyScoreExplanation.java

```diff
@@ -0,0 +1,252 @@
+    public static final ParseField ANOMALY_LENGTH = new ParseField("anomaly_length");
+    public static final ParseField SINGLE_BUCKET_IMPACT = new ParseField("single_bucket_impact");
+    public static final ParseField MULTI_BUCKET_IMPACT = new ParseField("multi_bucket_impact");
+    public static final ParseField ANOMALY_CHARACTERISTICS_IMPACT = new ParseField("anomaly_characteristics_impact");
+    public static final ParseField LOWER_CONFIDENCE_BOUND = new ParseField("lower_confidence_bound");
+    public static final ParseField TYPICAL_VALUE = new ParseField("typical_value");
+    public static final ParseField UPPER_CONFIDENCE_BOUND = new ParseField("upper_confidence_bound");
+    public static final ParseField HIGH_VARIANCE_PENALTY = new ParseField("high_variance_penalty");
+    public static final ParseField INCOMPLETE_BUCKET_PENALTY = new ParseField("incomplete_bucket_penalty");
+
+    public static final ObjectParser<AnomalyScoreExplanation, Void> STRICT_PARSER = createParser(false);
+    public static final ObjectParser<AnomalyScoreExplanation, Void> LENIENT_PARSER = createParser(true);
+
+    private static ObjectParser<AnomalyScoreExplanation, Void> createParser(Boolean ignoreUnknownFields) {
+        ObjectParser<AnomalyScoreExplanation, Void> parser = new ObjectParser<>(
+            ANOMALY_SCORE_EXPLANATION.getPreferredName(),
+            ignoreUnknownFields,
+            AnomalyScoreExplanation::new
+        );
+        parser.declareString(AnomalyScoreExplanation::setAnomalyType, ANOMALY_TYPE);
+        parser.declareInt(AnomalyScoreExplanation::setAnomalyLength, ANOMALY_LENGTH);
+        parser.declareInt(AnomalyScoreExplanation::setSingleBucketImpact, SINGLE_BUCKET_IMPACT);
+        parser.declareInt(AnomalyScoreExplanation::setMultiBucketImpact, MULTI_BUCKET_IMPACT);
+        parser.declareInt(AnomalyScoreExplanation::setAnomalyCharacteristicsImpact, ANOMALY_CHARACTERISTICS_IMPACT);
+        parser.declareDouble(AnomalyScoreExplanation::setLowerConfidenceBound, LOWER_CONFIDENCE_BOUND);
+        parser.declareDouble(AnomalyScoreExplanation::setTypicalValue, TYPICAL_VALUE);
+        parser.declareDouble(AnomalyScoreExplanation::setUpperConfidenceBound, UPPER_CONFIDENCE_BOUND);
+        parser.declareBoolean(AnomalyScoreExplanation::setHighVariancePenalty, HIGH_VARIANCE_PENALTY);
+        parser.declareBoolean(AnomalyScoreExplanation::setIncompleteBucketPenalty, INCOMPLETE_BUCKET_PENALTY);
+        return parser;
+    }
+
+    private String anomalyType;
+    private Integer anomalyLength;
+    private Integer singleBucketImpact;
+    private Integer multiBucketImpact;
+    private Integer anomalyCharacteristicsImpact;
+    private Double lowerConfidenceBound;
+    private Double typicalValue;
+    private Double upperConfidenceBound;
+    private Boolean highVariancePenalty;
+    private Boolean incompleteBucketPenalty;
+
+    public AnomalyScoreExplanation() {}
```

Comment:
It looks like this ctor is only used in tests and can be package private 

```suggestion
   AnomalyScoreExplanation() {}
```

Reply:
Fixed in a0e3327